src/utils/utils.js

```diff
@@ -31,3 +31,56 @@ export const getLongestStreak = (winlosses, matchString) => {
   }, "");
   return longestStreak.length;
 };
+
+export const compileWeeklySeeds = (owners, allMatchupsInYear) => {
+  const clonedOwners = [...owners];
+
+  allMatchupsInYear.forEach((matchupWeek, weekIndex) => {
+    const matchupMap = {};
+
+    // Group matchups by matchup_id
+    matchupWeek.forEach(team => {
+      const id = team.matchup_id;
+      if (!matchupMap[id]) matchupMap[id] = [];
+      matchupMap[id].push(team);
+    });
+
+    // Compare each matchup pair
+    Object.values(matchupMap).forEach(pair => {
+      if (pair.length === 2) {
+        const [teamA, teamB] = pair;
+
+        const ownerA = clonedOwners.find(o => o.rosterId === teamA.roster_id);
+        const ownerB = clonedOwners.find(o => o.rosterId === teamB.roster_id);
+
+        if (!ownerA || !ownerB) return;
+
+        ownerA.pointsCount += teamA.points;
+        ownerB.pointsCount += teamB.points;
+
+        if (teamA.points > teamB.points) {
+          ownerA.winCount += 1;
+        } else if (teamB.points > teamA.points) {
+          ownerB.winCount += 1;
+        }
+      }
+    });
+
+    // Rank all owners by winCount, then pointsCount
+    const rankings = [...clonedOwners]
+      .sort((a, b) => {
+        if (b.winCount === a.winCount) {
+          return b.pointsCount - a.pointsCount;
+        }
+        return b.winCount - a.winCount;
+      });
+
+    // Store weekly seeds
+    rankings.forEach((owner, i) => {
+      owner.seeds[`week${weekIndex + 1}`] = i + 1;
+    });
```

Comment:
I think the structure of this data could be tweaked to make this simpler. I think it makes sense for the `seeds` value to be an array that contains all the seeds. So `seeds: [1,1,2,1,4,etc]` instead of an object which lists each week as a key and then stores the seed as the value `seeds: {week1: 1, week2: 1, etc}`.

You'll need to update `App.js` so that the formattedUserData stores `seeds` as an array instead of an object, but then you could:

```js
rankings.forEach((owner, rank) => {
  const seed = rank + 1
  owner.seeds.push(seed)
});
```